src/_DataStructures_/LinkedList/index.js

```diff
@@ -121,30 +121,33 @@ class LinkedList {
     return node;
   }
 
-  removeAt(index) {
+  removeAt (index) {
     if (!this.head) {
-      return null;
+      return null
     }
-
-    if (index >= this.length()) {
+    if (index === 0) {
+      let returnNode=this.head;
+      this.head=this.head.next;
+      this.size -= 1;
+      return returnNode;
```

Comment:
return this.head from here

Reply:
```javascript
  if (!this.head) {
      this.head = node;
      this.tail = node;
      return node;
    }
```

If you do not make it `null` it will contain the full list